src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineWrappingInBlockCheck.java

```diff
@@ -0,0 +1,440 @@
+ * {@link TokenTypes#ANNOTATION_ARRAY_INIT ANNOTATION_ARRAY_INIT},
+ * {@link TokenTypes#ANNOTATION_DEF ANNOTATION_DEF},
+ * {@link TokenTypes#ARRAY_INIT ARRAY_INIT},
+ * {@link TokenTypes#CASE_GROUP CASE_GROUP},
+ * {@link TokenTypes#CLASS_DEF CLASS_DEF},
+ * {@link TokenTypes#CTOR_DEF CTOR_DEF},
+ * {@link TokenTypes#ENUM_CONSTANT_DEF ENUM_CONSTANT_DEF},
+ * {@link TokenTypes#ENUM_DEF ENUM_DEF},
+ * {@link TokenTypes#INSTANCE_INIT INSTANCE_INIT},
+ * {@link TokenTypes#INTERFACE_DEF INTERFACE_DEF},
+ * {@link TokenTypes#LAMBDA LAMBDA},
+ * {@link TokenTypes#LITERAL_CASE LITERAL_CASE},
+ * {@link TokenTypes#LITERAL_CATCH LITERAL_CATCH},
+ * {@link TokenTypes#LITERAL_DEFAULT LITERAL_DEFAULT},
+ * {@link TokenTypes#LITERAL_DO LITERAL_DO},
+ * {@link TokenTypes#LITERAL_ELSE LITERAL_ELSE},
+ * {@link TokenTypes#LITERAL_FINALLY LITERAL_FINALLY},
+ * {@link TokenTypes#LITERAL_FOR LITERAL_FOR},
+ * {@link TokenTypes#LITERAL_IF LITERAL_IF},
+ * {@link TokenTypes#LITERAL_NEW LITERAL_NEW},
+ * {@link TokenTypes#LITERAL_SWITCH LITERAL_SWITCH},
+ * {@link TokenTypes#LITERAL_SYNCHRONIZED LITERAL_SYNCHRONIZED},
+ * {@link TokenTypes#LITERAL_TRY LITERAL_TRY},
+ * {@link TokenTypes#LITERAL_WHILE LITERAL_WHILE},
+ * {@link TokenTypes#METHOD_DEF METHOD_DEF},
+ * {@link TokenTypes#OBJBLOCK OBJBLOCK},
+ * {@link TokenTypes#SLIST SLIST},
+ * {@link TokenTypes#STATIC_INIT STATIC_INIT}.
+ * </p>
+ *
+ * <p> By default the check will check the following statements:
+ *  {@link TokenTypes#ANNOTATION_DEF ANNOTATION_DEF},
+ *  {@link TokenTypes#CLASS_DEF CLASS_DEF},
+ *  {@link TokenTypes#ENUM_DEF ENUM_DEF},
+ *  {@link TokenTypes#INTERFACE_DEF INTERFACE_DEF}.
+ * </p>
+ *
+ * <p> An example of how to configure the check with default parameters is:
+ * </p>
+ *
+ * <pre>
+ * &lt;module name="EmptyLineWrappingInBlock"/&gt;
+ * </pre>
+ *
+ * <p> Example of declarations with empty line separator at the top of the block
+ * that is expected by the Check by default:
+ * </p>
+ *
+ * <pre>
+ * class Foo {
+ *
+ *     public int method() {
+ *
+ *         return 0;
+ *     }
+ * }
+ * </pre>
+ *
+ * <p> An example how to check wrapping of a {@link TokenTypes#CLASS_DEF CLASS_DEF} and
+ * {@link TokenTypes#INTERFACE_DEF INTERFACE_DEF} only:
+ * </p>
+ *
+ * <pre>
+ * &lt;module name="EmptyLineWrappingInBlock"&gt;
+ *    &lt;property name="tokens" value="CLASS_DEF, INTERFACE_DEF"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ *
+ * <p>
+ * An example how to allow {@link BlockWrapOption#NO_EMPTY_LINE no empty line} after the left curly
+ * brace nor before the right curly brace:
+ * </p>
+ *
+ * <pre>
+ * &lt;module name="EmptyLineWrappingInBlock"&gt;
+ *    &lt;property name="topSeparator" value="no-empty-line"/&gt;
+ *    &lt;property name="bottomSeparator" value="no-empty-line"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ *
+ * <p>
+ * An example how to demand {@link BlockWrapOption#EMPTY_LINE an empty line} after the left curly
+ * brace and before the right curly brace:
+ * </p>
```

Comment:
each config example should have java code block that show by trailing comment location of violation.
User love examples, it help better then any explanations.

Reply:
done.